src/blog/2024/02/postgresql-with-node-red.md

```diff
@@ -0,0 +1,218 @@
+2. Before connecting to your PostgreSQL database, ensure you have the following information ready. Later We will add environment variables to configure the PostgreSQL node to connect with the database:
+
+- Host: IP address or hostname of your PostgreSQL server.
+- Port: By default, PostgreSQL uses port 5432. Ensure this matches your PostgreSQL server configuration.
+- Database: The name of the PostgreSQL database you want to connect to.
+- User: username with the necessary privileges to access the specified database.
+- Password: corresponding password for the username.
+
+# Adding environment variables
+
+We have discussed many times in previous blogs that using environment variables is a best practice that prevents revealing confidential configuration details such as API keys, passwords, secret keys, etc directly in the flow. Therefore, In this section, we will add environment variables for Postgresql configuration details, for more details see  [Using Environment Variables in Node-RED](https://flowfuse.com/blog/2023/02/environment-variables-in-node-red/).
+
+1. Navigate to the instance's setting and then go to the environment section.
+2. Click on the `add variable` button and add variables for each of the configuration data that we discussed in the above section.
+3. Click on the save button and restart the instance by clicking on the top right `Action` button and selecting the restart option.
+
+!["Adding environment variables"](./images/postgresql_with_node-red_environment_variables.png "Adding environment variables")
+
+# Creating Table
+
+In this section, we will create a table in our database to store product data. Additionally, it's important to note that throughout this guide we will send an object containing the properties of configuration details with each request to the database. This approach allows us to use environment variables to secure our sensitive configuration details while connecting to the database.
+
+!["Setting pgConfig objects property"](./images/postgresql_with_node-red_change_node.png "Setting pgConfig objects property")
+
+1. Drag an Inject node onto canvas, and keep it unchanged.
+2. Drag the change node onto canvas, and set `msg.pgConfig` properties using the added environment variable as shown in the above image.
+3. Drag a Postgresql node onto Canvas, click on that node, and paste the following SQL command into the query input field (I have added comments for your understanding of sql commands)
+
+```sql
+-- This is an SQL INSERT statement used to add data into the product_data table.
+INSERT INTO product_data (id, name, price, stock)
+-- This line specifies the columns into which data will be inserted. The columns are id, name, price, and stock.
+-- It's important to match the columns in the same order as the values in the next line.
+VALUES ($1, $2, $3, $4);
+```
```

Comment:
I expected an `CREATE TABLE IF NOT EXISTS` statement here?

Reply:
I have separated operations for better understanding of readers. Therefore, I think the 'CREATE TABLE' statement should be updated with this approach, wdyt?